scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -471,24 +476,41 @@ def load_quarantine(self, file):
         # Load yaml into quarantine_yaml
         quarantine_yaml = scl.yaml_load_verify(file, self.quarantine_schema)
 
+
+        self.quarantine.platforms = quarantine_yaml.get('platforms', [])
+        self.quarantine.architectures = quarantine_yaml.get('architectures', [])
+        self.quarantine.simulation = quarantine_yaml.get('simulation', [])
+
         # Create quarantine_list with a product of the listed
         # platforms and scenarios for each entry in quarantine yaml
         quarantine_list = []
-        for quar_dict in quarantine_yaml:
+        for quar_dict in quarantine_yaml['tests']:
             if quar_dict['platforms'][0] == "all":
                 plat = self.platform_names
             else:
                 plat = quar_dict['platforms']
                 self.verify_platforms_existence(plat, "quarantine-list")
             comment = quar_dict.get('comment', "NA")
-            quarantine_list.append([{".".join([p, s]): comment}
-                                   for p in plat for s in quar_dict['scenarios']])
+            scenarios = []
+            for _, ts in self.testsuites.items():
+                scenarios.append(ts.id)
+            q_scenarios = []
+            for s in quar_dict['scenarios']:
+                if '.*' in s:
```

Comment:
Nice feature. But maybe instead of checking for '.*' would be better to use raw string ( r'pattern', e.g. r'kernel\.common.*')?
In current solution, if you add '.*' to tests with dots, every dot will be taken by a regex as any letter.

Reply:
I guess I can remove the whole thing and allow any type of regex, so this becomes:

```
+            for s in quar_dict['scenarios']:
+                r = re.compile(s)
+                q_scenarios.extend(list(filter(r.match, scenarios)))
```